src/Microsoft.DotNet.Build.CloudTestTasks/WriteBuildStatsJson.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    /// <summary>
+    /// Used to write build stats to a JSON file.
+    /// </summary>
+    public class WriteTestBuildStatsJson : Task
+    {
+        /// <summary>
+        /// Helper class to be serialized to JSON.
+        /// </summary>
+        class TestBuildStatsJson
+        {
+            internal class FailedProject
+            {
+                private string m_name;
+
+                public FailedProject(string name)
+                {
+                    m_name = name;
+                }
+
+                public string ProjectName { get { return m_name; } }
+            }
+
+            private List<string> m_aggregates;
+            private List<FailedProject> m_failedProjects;
+            private string m_logUri;
+            private int m_builtCount;
+            private int m_testCount;
+
+            public TestBuildStatsJson(ITaskItem[] correlationIDs, string logUri, int builtCount, ITaskItem[] buildFailures, int testCount)
+            {
+                m_aggregates = new List<string>();
+                m_failedProjects = new List<FailedProject>();
+
+                foreach (var correlationId in correlationIDs)
+                    m_aggregates.Add(correlationId.ItemSpec);
+
+                foreach (var buildFailure in buildFailures)
+                    m_failedProjects.Add(new FailedProject(buildFailure.ItemSpec));
+
+                m_logUri = logUri;
+                m_builtCount = builtCount;
+                m_testCount = testCount;
+            }
+
+            public List<string> Aggregates { get { return m_aggregates; } }
```

Comment:
I'm not sure what compiler this project uses, but you could just have get only properties that you assign in the constructor... should make the file a bit shorter and easier to fit on a screen. :-)